src/sagemaker_sklearn_extension/feature_extraction/sequences.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#      http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_array, check_is_fitted
+from tsfresh import extract_features
+from tsfresh.feature_extraction import ComprehensiveFCParameters
+from tsfresh.utilities.dataframe_functions import impute
+
+from sagemaker_sklearn_extension.preprocessing.data import RobustStandardScaler
+
+
+class TSFlattener(BaseEstimator, TransformerMixin):
+    """Convert lists of strings of varying length into an np.array.
+
+    The input is a collection of lists of strings, with each string containing a sequence of comma-separate numbers.
+    TSFlattener extracts the numerical values contained in these strings and pads shorter sequences (if any) with NaNs,
+    so that all sequences match the length of the longest sequence. TSFlattener returns a single np.array as output.
+
+    Any missing values (represented either by a NaN, inf or empty string) are converted to np.nans.
+
+    Example:
+    Input = [["1,, 3, 44"],
+             ["11, 111"],
+             ["NaN, , 1, NaN"]]
+    Output = [[1, np.nan, 3, 44],
+              [11, 111, np.nan, np.nan],
+              [np.nan, np.nan, 1, np.nan]
+              ]
+    """
+
+    def fit(self, X, y=None):
+        check_array(X, dtype=None, force_all_finite="allow-nan")
+        return self
+
+    def transform(self, X, y=None):
+        """Extract numerical values from strings of comma-separated numbers and returns an np.array.
+
+        - If some sequences are shorter than the longest sequence, these are padded with np.nans.
+        - Any NaN, inf or empty string is converted to a np.nan.
+
+        Parameters
+        ----------
+        X : lists of strings
+
+        Returns
+        -------
+        X : np.array
+
+        """
+        X = check_array(X, dtype=None, force_all_finite="allow-nan")
+
+        numeric_sequences = []
+        for string_sequence in X:
+            assert len(string_sequence) == 1, (
+                f"TSFlattener can process a single sequence column at a time, "
+                f"but it was given {len(string_sequence)} sequence columns."
+            )
+            numeric_sequence = [float(s) if (s and not s.isspace()) else np.nan for s in string_sequence[0].split(",")]
```

Comment:
this is problematic. It will crash the system if we have a single row with non-numerics

"1,2,a,3,4"

Instead, it should turn anything that can't be converted to a float to np.nan

Reply:
Done, and added a test for the non-numeric case.